### PR TITLE
ENH: support multi-stream header_io

### PR DESCRIPTION
**Pull request by CJ-Wright**
among other things

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@danielballan here are some fixes

**Comment by CJ-Wright** [comment]
@ordirules This should be ready for some review.

**Comment by ordirules** [comment]
looks good, thanks for fixing a bug I introduced... (that circular newdatakeys logic was mine, not @danielballan 's...)

there's just one thing you need to change, submitted it as a PR, feel free to merge it or write it in.
https://github.com/CJ-Wright/databroker/pull/2

thanks!

File: databroker/broker.py [hunk 1]
```diff
@@ -1068,10 +1074,10 @@ def inner(stream):
 
 def header_io(db_in, db_out):
     def outer(f):
-        def inner(header):
+        def inner(headers):
```
Comment (danielballan) [inline databroker/broker.py]: Actually there is no need for this. `restream` accepts a list of headers and returns one long stream, so this use case should work as is.

Comment (CJ-Wright) [inline databroker/broker.py]: except I don't one long stream I need them splayed out

File: databroker/broker.py [hunk 2]
```diff
@@ -1075,6 +1075,8 @@ def inner(stream):
 def header_io(db_in, db_out):
     def outer(f):
         def inner(headers):
+            if not isinstance(headers, list):
```
Comment (ordirules) [inline databroker/broker.py]: :+1:

File: databroker/broker.py (hunk 1 above)
Comment (danielballan) [inline databroker/broker.py]: Fair enough, let's see where that goes.

I can merge this as is if you (1) rebase and (2) change `not isinstance(headers, list)` to `hasattr(headers, 'items')`. We should also headers to be any iterable collection, not strictly list. Looking for 'items' is a nice way to check if it's a dict (dict-like).

Comment (CJ-Wright) [inline databroker/broker.py]: We should discuss how to handle multiple input streams if you have time at some point (are they handed in as a tuple, args, dict, or kwargs?)

**Pull request closed without merging**